dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/Responses/Converters/ResponseConverterExtensions.cs

```diff
@@ -0,0 +1,193 @@
+        return new FunctionToolCallItemResource(
+            id: id,
+            status: FunctionToolCallItemResourceStatus.Completed,
+            callId: functionCallContent.CallId,
+            name: functionCallContent.Name,
+            arguments: JsonSerializer.Serialize(functionCallContent.Arguments, jsonSerializerOptions)
+        );
+#pragma warning restore IL2026, IL3050
+    }
+
+    /// <summary>
+    /// Converts FunctionResultContent to a FunctionToolCallOutputItemResource.
+    /// </summary>
+    /// <param name="functionResultContent">The function result content to convert.</param>
+    /// <param name="id">The ID to assign to the resource.</param>
+    /// <returns>A FunctionToolCallOutputItemResource.</returns>
+    public static FunctionToolCallOutputItemResource ToFunctionToolCallOutputItemResource(
+        this FunctionResultContent functionResultContent,
+        string id)
+    {
+        var output = functionResultContent.Exception is not null
+            ? $"{functionResultContent.Exception.GetType().Name}(\"{functionResultContent.Exception.Message}\")"
+            : $"{functionResultContent.Result?.ToString() ?? "(null)"}";
+        return new FunctionToolCallOutputItemResource(
+            id: id,
+            status: FunctionToolCallOutputItemResourceStatus.Completed,
+            callId: functionResultContent.CallId,
+            output: output
+        );
+    }
+
+    /// <summary>
+    /// Converts UsageDetails to ResponseUsage.
+    /// </summary>
+    /// <param name="usage">The usage details to convert.</param>
+    /// <returns>A ResponseUsage object, or null if usage is null.</returns>
+    public static ResponseUsage? ToResponseUsage(this UsageDetails? usage)
+    {
+        if (usage == null)
+        {
+            return null;
+        }
+
+        var cachedTokens = usage.AdditionalCounts?.TryGetValue("InputTokenDetails.CachedTokenCount", out var cachedInputToken) ?? false
+            ? (int)cachedInputToken
+            : 0;
+        var reasoningTokens =
+            usage.AdditionalCounts?.TryGetValue("OutputTokenDetails.ReasoningTokenCount", out var reasoningToken) ?? false
+                ? (int)reasoningToken
+                : 0;
+
+        return new ResponseUsage
+        {
+            InputTokens = (int)(usage.InputTokenCount ?? 0),
+            InputTokensDetails = new InputTokensDetails { CachedTokens = cachedTokens },
+            OutputTokens = (int)(usage.OutputTokenCount ?? 0),
+            OutputTokensDetails = new OutputTokensDetails { ReasoningTokens = reasoningTokens },
+            TotalTokens = (int)(usage.TotalTokenCount ?? 0)
+        };
+    }
+
+    /// <summary>
+    /// Converts AIContent to ItemContent.
+    /// </summary>
+    /// <param name="content">The AI content to convert.</param>
+    /// <returns>An ItemContent object, or null if the content cannot be converted.</returns>
+    public static ItemContent? ToItemContent(this AIContent content)
```

Comment:
This is a lot less complete than ConvertToItemContent later on... why do we have two?

Reply:
Fixed